Automattic/studio
# Add language switcher in user settings modal

- Closes https://github.com/Automattic/studio/issues/25

## Proposed Changes

- Add language picker in user settings for authenticated and non authenticated users, which allows changing the language on the fly.
- Add language to AppData user data file
- Remove locale from AppGlobals and refactor locale code

The only downside is that the user needs to be logged in to open the user settings.

## Testing Instructions



- Run `npm start`
- Observe the App starts with the computer's language
- Login to Studio with your WPcom account
- Click in your profile picture in the left sidebar
- Change the language to any language, even RTL
- Observe the interface changes immediately
- Close the app
- Run `npm start`
- Observe the App starts with the latest language you selected
- Open the settings
- Log out
- Press `cmd+.` on Mac or use the menu bar to open the settings modal
- Observe the language picker is still in place and works as expected.


https://github.com/user-attachments/assets/dc252326-8c0b-4a82-86b4-aed0f3619431


No Auth:

<img width="1012" alt="Screenshot 2024-09-02 at 11 43 58" src="https://github.com/user-attachments/assets/be286efb-6dba-4323-a4ae-7d680efd9ac7">

Auth:

<img width="1012" alt="Screenshot 2024-09-02 at 11 43 35" src="https://github.com/user-attachments/assets/92e4e95f-c8dc-4708-b944-6681397c9378">



## Pre-merge Checklist
